File: perl/sub_as_ref.pl
#!/usr/bin/perl

use strict;
use warnings;

my %values=(
        "id" => \&get_timestamp,
        "name" => \&do_stuff_to_name,
        "age" => \&lie_about_age);

sub get_timestamp{
        return time();
}

sub do_stuff_to_name{
        my $name=$_[0];
        return "aaa ".$name." aaa";
}

sub lie_about_age{
        my $name=$_[0];
        return "$name are ".int(rand(100))." ani";
}

for my $attr (keys %values){
        print &{$values{$attr}}("gigi")."\n";
}

File: python/sub_as_ref.py
#!/usr/bin/python

def get_timestamp(name):
	return 1

def do_stuff_to_name(name):
	return name

def lie_about_age(name):
	return name

values={
	"id" : get_timestamp,
	"name" : do_stuff_to_name,
	"age" : lie_about_age
}

for key in values.keys():
	print(values[key]("gigi"))

